### PR TITLE
refactor: use mux pkg from salt

### LINKED ISSUE #164: bug: graceful shutdown is not so graceful
**Is your feature request related to a problem? Please describe.**
Compass panics on graceful shutdown with below stacktrace
```
panic: Drain() is not implemented

goroutine 56 [running]:
google.golang.org/grpc/internal/transport.(*serverHandlerTransport).Drain(0xc000e340a0?)
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.46.0/internal/transport/handler_server.go:439 +0x27
google.golang.org/grpc.(*Server).GracefulStop(0xc000821880)
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:1765 +0x3f4
github.com/odpf/compass/internal/server.Serve.func1()
	/home/runner/work/compass/compass/internal/server/server.go:157 +0x1bb
created by github.com/odpf/compass/internal/server.Serve
	/home/runner/work/compass/compass/internal/server/server.go:143 +0xaae
```

Checking from this [issue](https://github.com/grpc/grpc-go/issues/1384#issuecomment-317124531), this looks like the expected behaviour when calling `grpcServer.GracefulStop()`.

Compass calls `grpcServer.GracefulStop()` as part of `graceful shutdown process` as seen [here](https://github.com/odpf/compass/blob/main/internal/server/server.go#L157).

Compass should not panic on graceful shutdown.

**Describe the solution you'd like**

### 1. Remove calling GracefulStop

We can remove calling `grpcServer.GracefulStop()` as we are actually already handling shutdown gracefully.

### 2. Catch panic

Another approach if we just catch that panic and recover properly.

### PR DESCRIPTION
**Pull request by sudo-suhas**

### CODE REVIEW

File: core/asset/service.go [hunk 1]
```diff
@@ -76,6 +77,7 @@ func (s *Service) DeleteAsset(ctx context.Context, id string) error {
 }
 
 func (s *Service) GetAssetByID(ctx context.Context, id string) (ast Asset, err error) {
+	time.Sleep(10 * time.Second)
```
Comment (sudo-suhas) [inline core/asset/service.go:80]: `FOR TESTING. DELETE BEFORE MERGE!!!`

**Comment by sudo-suhas** [comment]
@spy16 am still able to reproduce the panic locally. Can you check if I am doing something wrong?

After introducing the artificial delay for `GetAssetByID` response, I attempt to stop the server with `ctrl+c` from the command line:

```
$ # start the server from terminal 1
$ go run main.go serve
INFO[0000] compass starting                              version=
...
INFO[0000] Starting server                               addr="localhost:8080"

$ # start the request from terminal 2
$ grpcurl -plaintext -vv \
  -H 'Compass-User-UUID: test' \
  -H 'Compass-User-Email: test@test.com' \
  -d '{"id": "urn:caramlstore:core.s.ng.feast.golabs.io:ml_feature_table:merchant_similarity_dummy"}' \
  localhost:8080 odpf.compass.v1beta1.CompassService/GetAssetByID

$ # press ctrl+c in terminal 1
^CWARN[0005] closing db...
panic: Drain() is not implemented

goroutine 1 [running]:
google.golang.org/grpc/internal/transport.(*serverHandlerTransport).Drain(0x140000021a0?)
	/Users/skaranth/dev/work/go/pkg/mod/google.golang.org/grpc@v1.49.0/internal/transport/handler_server.go:439 +0x2c
google.golang.org/grpc.(*Server).GracefulStop(0x140006cad20)
	/Users/skaranth/dev/work/go/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:1787 +0x45c
github.com/odpf/salt/mux.(*muxServer).Serve(0x14000749320, {0x101a05a70, 0x14000051e40}, {0x1400068e590, 0xe})
	/Users/skaranth/dev/work/go/pkg/mod/github.com/odpf/salt@v0.2.4/mux/mux.go:72 +0x278
github.com/odpf/salt/mux.Serve({0x101a05a70, 0x14000051e40}, {0x1400068e590, 0xe}, {0x1400088ef40, 0x3, 0x11?})
	/Users/skaranth/dev/work/go/pkg/mod/github.com/odpf/salt@v0.2.4/mux/mux.go:34 +0xb4
github.com/odpf/compass/internal/server.Serve({0x101a05a70, 0x14000051e40}, {{0x1400037c2a0, 0x9}, 0x1f90, {0x10178193e, 0xe}, {{0x14000048600, 0x11}, {0x101805598, ...}, ...}}, ...)
	/Users/skaranth/dev/work/odpf/compass/internal/server/server.go:133 +0x960
github.com/odpf/compass/cli.runServer({{0x1400037c0d8, 0x4}, {0x0, {0x1400037c120, 0xe}, {0x1400037c148, 0x7}, 0x3ff0000000000000, {0x10174361d, 0x1}, ...}, ...})
	/Users/skaranth/dev/work/odpf/compass/cli/serve.go:124 +0xc50
github.com/odpf/compass/cli.cmdServe.func1(0x140007ff900?, {0x10254b810?, 0x0?, 0x0?})
	/Users/skaranth/dev/work/odpf/compass/cli/serve.go:49 +0xa0
github.com/spf13/cobra.(*Command).execute(0x140007ff900, {0x10254b810, 0x0, 0x0})
	/Users/skaranth/dev/work/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856 +0x4c4
github.com/spf13/cobra.(*Command).ExecuteC(0x1024f56e0)
	/Users/skaranth/dev/work/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x354
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/skaranth/dev/work/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/odpf/compass/cli.Execute()
	/Users/skaranth/dev/work/odpf/compass/cli/cli.go:67 +0x200
main.main()
	/Users/skaranth/dev/work/odpf/compass/main.go:6 +0x1c
exit status 2
```

**Comment by sudo-suhas** [comment]
@StewartJingga although this won't fix the bug, should we still go ahead and merge this change?

**Comment by ravisuhag** [comment]
I think yes.

File: core/asset/service.go (hunk 1 above)
Comment (sudo-suhas) [inline core/asset/service.go:80]:
```suggestion
```

**Pull request closed without merging**